frontend/src/components/utils/DatamapsExtension.tsx

```diff
@@ -33,9 +29,10 @@ export default class Datamap extends React.Component<Props> {
     this.drawMap();
   }
 
-  componentDidUpdate() {
-    this.clear();
-    this.drawMap();
+  componentDidUpdate(prevProps: Props) {
+    if (this.props.bubbles.length !== prevProps.bubbles.length) {
+      this.drawMap();
+    }
```

Comment:
1. If you don't want to re-render, it makes sense to implement `shouldComponentUpdate`
2. Are you sure it is safe to draw without clearing and also skip some drawing if the length of the arrays is the same? What if one node goes offline and another one goes online (the length will be the same as far as I can tell)?

Reply:
not very sure, need your help here. @frol